codex-rs/core/src/shell.rs

```diff
@@ -131,6 +134,178 @@ fn strip_bash_lc(command: &Vec<String>) -> Option<String> {
     }
 }
 
+/// Returns a copy of `command` that is easier for humans to read when rendered
+/// in activity logs and error messages.
+///
+/// For shell wrappers such as `bash -lc "cd repo && git status"` this removes
+/// the leading `cd` segment so the displayed command mirrors what will actually
+/// run inside the working directory. When the command is provided as plain
+/// tokens (`["cd", "repo", "&&", "git", "status"]`) the prefix is stripped in
+/// the same way. Commands that do not look like `cd <path> && ...` are returned
+/// unchanged.
+pub fn prettify_command_for_display(command: &[String]) -> Vec<String> {
+    if let Some((script_idx, script)) = extract_shell_script(command)
+        && let Some(split) = split_cd_prefix_from_script(script)
+    {
+        let mut display = command.to_vec();
+        display[script_idx] = split.remainder;
+        return display;
+    }
+
+    if let Some(remainder) = strip_cd_prefix_from_tokens(command) {
```

Comment:
Again, this does not seem correct. `command` is overloaded to mean two different things here (args to `exec()` versus a split string of shell). I'm not comfortable confusing these two concepts.